piff/gaussian_model.py

```diff
@@ -62,21 +66,34 @@ def fit(self, star):
         flux = mom.moments_amp
         center = mom.moments_centroid
 
-        # Also need to compute chisq
-        prof = self.getProfile(params) * flux
-        offset = center - image.trueCenter()
-        model_image = prof.drawImage(image.copy(), method='no_pixel', offset=offset)
+        # # Also need to compute chisq
+        # prof = self.getProfile(params) * flux
+        # offset = center - image.trueCenter()
+        # model_image = prof.drawImage(image.copy(), method='no_pixel', offset=offset)
+        # chisq = numpy.std(image.array - model_image.array)
+        # dof = numpy.count_nonzero(weight.array) - 6
+
+        fit = StarFit(params, flux=flux, center=center-image_pos)
+        model_star = Star(star.data, fit)
+
+        # compute chisq
+        model_star = self.draw(model_star)
+        model_image = model_star.data.getImage()[0]
         chisq = numpy.std(image.array - model_image.array)
         dof = numpy.count_nonzero(weight.array) - 6
-
-        fit = StarFit(params, flux=flux, center=center-image_pos, chisq=chisq, dof=dof)
+        fit = StarFit(params, flux=model_star.fit.flux, center=model_star.fit.center, chisq=chisq, dof=dof)
         return Star(star.data, fit)
 
+
+    def update(self, logger=None, **kwargs):
+        self.kwargs.update(kwargs)
```

Comment:
I don't think you want to take any arbitrary kwargs here do you?  Only `background` is valid here.  (Although as I mentioned, I don't think we want that here, but I see why you did it for the hyper parameter tests...)  In general, you should list the valid kwargs individually so that if someone passes something that the class won't understand, they'll get a `ValueError` rather than silently updating the kwargs with something the class will just ignore.

But also, I'm not sure I like this way of keeping track of the hyper parameters.  I had been imagining these kwargs to be just used for the I/O to know how to build objects when we read them in.  I think if we're envisioning Models and Interps with hyper-parameters, we should have a struct that is designed for them specifically rather than piggy-backing on the I/O functionality.  

Maybe it's ok (since it's not invalid for these hyper-parameters to also be construction arguments), but it would require making sure that classes always look into the `self.kwargs` dict for the value of things rather than having attributes, which would be more natural.

So maybe for classes that have hyper parameters, you can have `updateHyper` which would update the hyper parameters directly.  And then not have such a function for other Models and Interps that wouldn't know what to do with them.  This seems better than silently passing as you now do for the default `update` implementation.